### PR TITLE
Debug4jConfig

### PR DESCRIPTION
**Pull request by lopelopelope**
## What changes were proposed in this pull request?
provide the configure file to run debug mode

## Related links or issues (optional)
fixed https://github.com/intel-analytics/BigDL/issues/1521

### CODE REVIEW

**Comment by yiheng** [comment]
How to use this new file?

File: spark/log4jDebug.xml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8" ?>
```
Comment (yiheng) [inline spark/log4jDebug.xml]: I think such file should be placed in the resource folder. Please check this page: https://maven.apache.org/guides/introduction/introduction-to-the-standard-directory-layout.html

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/example/MLPipeline/DLClassifierLeNet.scala [hunk 2]
```diff
@@ -37,7 +37,7 @@ import org.apache.spark.sql.SQLContext
 object DLClassifierLeNet {
 
   LoggerFilter.redirectSparkInfoLogs()
-  Logger.getLogger("com.intel.analytics.bigdl.optim").setLevel(Level.INFO)
+//  Logger.getLogger("com.intel.analytics.bigdl.optim").setLevel(Level.INFO)
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/example/MLPipeline/DLClassifierLeNet.scala]: If you want to remove code, remove it. NEVER comment it to remove code.

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/example/lenetLocal/Predict.scala [hunk 3]
```diff
@@ -27,7 +27,7 @@ object Predict {
   Logger.getLogger("org").setLevel(Level.ERROR)
   Logger.getLogger("akka").setLevel(Level.ERROR)
   Logger.getLogger("breeze").setLevel(Level.ERROR)
-  Logger.getLogger("com.intel.analytics.bigdl.optim").setLevel(Level.INFO)
+ // Logger.getLogger("com.intel.analytics.bigdl.optim").setLevel(Level.INFO)
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/example/lenetLocal/Predict.scala]: remove it

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/example/lenetLocal/Test.scala [hunk 4]
```diff
@@ -27,7 +27,7 @@ object Test {
   Logger.getLogger("org").setLevel(Level.ERROR)
   Logger.getLogger("akka").setLevel(Level.ERROR)
   Logger.getLogger("breeze").setLevel(Level.ERROR)
-  Logger.getLogger("com.intel.analytics.bigdl.optim").setLevel(Level.INFO)
+ // Logger.getLogger("com.intel.analytics.bigdl.optim").setLevel(Level.INFO)
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/example/lenetLocal/Test.scala]: remove it

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/example/lenetLocal/Train.scala [hunk 5]
```diff
@@ -29,7 +29,7 @@ import org.apache.log4j.{Level, Logger}
 
 object Train {
   LoggerFilter.redirectSparkInfoLogs()
-  Logger.getLogger("com.intel.analytics.bigdl.optim").setLevel(Level.INFO)
+  // Logger.getLogger("com.intel.analytics.bigdl.optim").setLevel(Level.INFO)
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/example/lenetLocal/Train.scala]: remove it

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/models/autoencoder/Train.scala [hunk 6]
```diff
@@ -49,7 +49,7 @@ object Train {
   Logger.getLogger("org").setLevel(Level.ERROR)
   Logger.getLogger("akka").setLevel(Level.ERROR)
   Logger.getLogger("breeze").setLevel(Level.ERROR)
-  Logger.getLogger("com.intel.analytics.bigdl.optim").setLevel(Level.INFO)
+  // Logger.getLogger("com.intel.analytics.bigdl.optim").setLevel(Level.INFO)
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/models/autoencoder/Train.scala]: remove it

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/models/inception/Test.scala [hunk 7]
```diff
@@ -29,7 +29,7 @@ object Test {
   Logger.getLogger("org").setLevel(Level.ERROR)
   Logger.getLogger("akka").setLevel(Level.ERROR)
   Logger.getLogger("breeze").setLevel(Level.ERROR)
-  Logger.getLogger("com.intel.analytics.bigdl.optim").setLevel(Level.INFO)
+//  Logger.getLogger("com.intel.analytics.bigdl.optim").setLevel(Level.INFO)
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/models/inception/Test.scala]: remove it

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/models/inception/Train.scala [hunk 8]
```diff
@@ -24,7 +24,7 @@ import org.apache.spark.SparkContext
 
 object TrainInceptionV1 {
   LoggerFilter.redirectSparkInfoLogs()
-  Logger.getLogger("com.intel.analytics.bigdl.optim").setLevel(Level.INFO)
+//  Logger.getLogger("com.intel.analytics.bigdl.optim").setLevel(Level.INFO)
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/models/inception/Train.scala]: remove it

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/models/lenet/Test.scala [hunk 9]
```diff
@@ -30,7 +30,7 @@ object Test {
   Logger.getLogger("org").setLevel(Level.ERROR)
   Logger.getLogger("akka").setLevel(Level.ERROR)
   Logger.getLogger("breeze").setLevel(Level.ERROR)
-  Logger.getLogger("com.intel.analytics.bigdl.optim").setLevel(Level.INFO)
+//  Logger.getLogger("com.intel.analytics.bigdl.optim").setLevel(Level.INFO)
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/models/lenet/Test.scala]: remove it

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/models/lenet/Train.scala [hunk 10]
```diff
@@ -28,7 +28,7 @@ import org.apache.spark.SparkContext
 
 object Train {
   LoggerFilter.redirectSparkInfoLogs()
-  Logger.getLogger("com.intel.analytics.bigdl.optim").setLevel(Level.INFO)
+ // Logger.getLogger("com.intel.analytics.bigdl.optim").setLevel(Level.INFO)
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/models/lenet/Train.scala]: same

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/models/resnet/Test.scala [hunk 11]
```diff
@@ -29,7 +29,7 @@ object Test {
   Logger.getLogger("org").setLevel(Level.ERROR)
   Logger.getLogger("akka").setLevel(Level.ERROR)
   Logger.getLogger("breeze").setLevel(Level.ERROR)
-  Logger.getLogger("com.intel.analytics.bigdl.optim").setLevel(Level.INFO)
+//  Logger.getLogger("com.intel.analytics.bigdl.optim").setLevel(Level.INFO)
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/models/resnet/Test.scala]: same

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/models/resnet/Train.scala [hunk 12]
```diff
@@ -27,7 +27,7 @@ import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric._
 
 object Train {
   LoggerFilter.redirectSparkInfoLogs()
-  Logger.getLogger("com.intel.analytics.bigdl.optim").setLevel(Level.INFO)
+//  Logger.getLogger("com.intel.analytics.bigdl.optim").setLevel(Level.INFO)
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/models/resnet/Train.scala]: same

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/models/rnn/Test.scala [hunk 13]
```diff
@@ -34,7 +34,7 @@ object Test {
   Logger.getLogger("org").setLevel(Level.ERROR)
   Logger.getLogger("akka").setLevel(Level.ERROR)
   Logger.getLogger("breeze").setLevel(Level.ERROR)
-  Logger.getLogger("com.intel.analytics.bigdl.optim").setLevel(Level.INFO)
+ // Logger.getLogger("com.intel.analytics.bigdl.optim").setLevel(Level.INFO)
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/models/rnn/Test.scala]: same

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/models/rnn/Train.scala [hunk 14]
```diff
@@ -33,7 +33,7 @@ object Train {
   Logger.getLogger("org").setLevel(Level.ERROR)
   Logger.getLogger("akka").setLevel(Level.ERROR)
   Logger.getLogger("breeze").setLevel(Level.ERROR)
-  Logger.getLogger("com.intel.analytics.bigdl.optim").setLevel(Level.INFO)
+//  Logger.getLogger("com.intel.analytics.bigdl.optim").setLevel(Level.INFO)
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/models/rnn/Train.scala]: same

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/models/vgg/Test.scala [hunk 15]
```diff
@@ -29,7 +29,7 @@ object Test {
   Logger.getLogger("org").setLevel(Level.ERROR)
   Logger.getLogger("akka").setLevel(Level.ERROR)
   Logger.getLogger("breeze").setLevel(Level.ERROR)
-  Logger.getLogger("com.intel.analytics.bigdl.optim").setLevel(Level.INFO)
+  // Logger.getLogger("com.intel.analytics.bigdl.optim").setLevel(Level.INFO)
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/models/vgg/Test.scala]: same

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/models/vgg/Train.scala [hunk 16]
```diff
@@ -27,7 +27,7 @@ import org.apache.spark.SparkContext
 
 object Train {
   LoggerFilter.redirectSparkInfoLogs()
-  Logger.getLogger("com.intel.analytics.bigdl.optim").setLevel(Level.INFO)
+ // Logger.getLogger("com.intel.analytics.bigdl.optim").setLevel(Level.INFO)
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/models/vgg/Train.scala]: same

File: spark/log4jDebug.xml (hunk 1 above)
Comment (yiheng) [reply spark/log4jDebug.xml]: put the file to resource folder

Comment (yiheng) [reply spark/log4jDebug.xml]: why we need this file?

**Comment by yiheng** [comment]
Update the markdown document to show people how to display the debug info of distributOptiizer

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/example/MLPipeline/DLClassifierLeNet.scala (hunk 2 above)
Comment (yiheng) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/example/MLPipeline/DLClassifierLeNet.scala]: Are you sure you can remove the log level? What's the log looks like when you run the example after remove it?

**Comment by lopelopelope** [comment]
@yiheng I have put the configuration file and its usage in doc using this PR https://github.com/intel-analytics/bigdl-doc/pull/2/files. I also deleted the codes instead of  commenting them as you wish so that the configuration file will work properly.

File: docs/docs/getting-started.md [hunk 17]
```diff
@@ -101,8 +101,47 @@ BigDL provides a convinient way to monitor/visualize your training progress. It
 ## **Tuning**
 
 There're several strategies that may be useful when tuning an optimization. 
+ * Change the learning Rate Schedule in SGD. Refer to [SGD docs](APIdocs/Optimizers/Optim-Methods.md#sgd) for details. 
+ * If overfit is seen, try use Regularization. Refer to [Regularizers](APIdocs/Regularizers.md). 
+ * Try change the initialization methods. Refer to [Initailizers](APIdocs/Initializers.md).
+ * Try Adam or Adagrad at the first place. If they can't achive a good score, use SGD and find a proper learning rate schedule - it usually takes time, though. RMSProp is recommended for RNN models. Refer to [Optimization Algorithms](APIdocs/Optimizers/Optim-Methods.md) for a list of supported optimization methods. 
+ 
+ 
+## Use log4j properity file to investigate issue using Debug log
```
Comment (yiheng) [inline docs/docs/getting-started.md]: move this section to https://bigdl-project.github.io/master/#ProgrammingGuide/optimization/

Comment (yiheng) [reply docs/docs/getting-started.md]: ## Performance Tunning
For performance investigation, BigDL records the time-consuming distribution on each node for each step(e.g. sync weight, computing). The information can be displayed in the driver log. By default, it is suspended. To turn it on, please follow these steps:

1. Prepare a log4j property file
```
# Root logger option
log4j.rootLogger=INFO, stdout

# Direct log messages to stdout
log4j.appender.stdout=org.apache.log4j.ConsoleAppender
log4j.appender.stdout.Target=System.out
log4j.appender.stdout.layout=org.apache.log4j.PatternLayout
log4j.appender.stdout.layout.ConversionPattern=%d{yyyy-MM-dd HH:mm:ss} %-5p %c{1}:%L - %m%n

log4j.logger.com.intel.analytics.bigdl.optim=DEBUG,myappender
log4j.addivity.com.intel.analytics.bigdl.optim=false

log4j.appender.myappender=org.apache.log4j.ConsoleAppender
log4j.appender.myappender.Target=System.out
log4j.appender.myappender.layout=org.apache.log4j.PatternLayout
log4j.appender.myappender.layout.ConversionPattern=%d{yyyy-MM-dd HH:mm:ss} %-5p %c{1}:%L - %m%n
```

2. Add an option to your spark-submit command
```
--conf "spark.driver.extraJavaOptions=-Dlog4j.configuration=file:where_is_your_log4j_file."
```

File: docs/docs/getting-started.md [hunk 18]
```diff
@@ -101,8 +101,9 @@ BigDL provides a convinient way to monitor/visualize your training progress. It
 ## **Tuning**
 
 There're several strategies that may be useful when tuning an optimization. 
-
- * Change the learning Rate Schedule in SGD. Refer to [SGD docs](APIGuide/Optimizers/Optim-Methods.md#sgd) for details. 
- * If overfit is seen, try use Regularization. Refer to [Regularizers](APIGuide/Regularizers.md). 
- * Try change the initialization methods. Refer to [Initailizers](APIGuide/Initializers.md).
- * Try Adam or Adagrad at the first place. If they can't achive a good score, use SGD and find a proper learning rate schedule - it usually takes time, though. RMSProp is recommended for RNN models. Refer to [Optimization Algorithms](APIGuide/Optimizers/Optim-Methods.md) for a list of supported optimization methods. 
+ * Change the learning Rate Schedule in SGD. Refer to [SGD docs](APIdocs/Optimizers/Optim-Methods.md#sgd) for details.
```
Comment (yiheng) [inline docs/docs/getting-started.md]: why this file is changed?

**Pull request merged**